framework/src/base/MooseServer.C

```diff
@@ -0,0 +1,529 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MooseServer.h"
+#include "Moose.h"
+#include "InputParameters.h"
+#include "AppFactory.h"
+#include "pcrecpp.h"
+#include <cstdio>
+
+bool
+MooseServer::setup(MooseApp * moose_app)
+{
+  bool pass = true;
+
+  // cache a pointer to the application that set up this server instance
+
+  this->moose_app = moose_app;
+
+  this->is_setup = true;
+
+  return pass;
+}
+
+bool
+MooseServer::parseDocumentForDiagnostics(wasp::DataArray & diagnosticsList)
+{
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // strip prefix from document path if it exists and append file suffix
+
+  std::string parse_file_path = this->document_path + ".tmpls";
+
+  if (parse_file_path.rfind(wasp::lsp::m_uri_prefix, 0) == 0)
+  {
+    parse_file_path.erase(0, std::string(wasp::lsp::m_uri_prefix).size());
+  }
+
+  // write file content to temporary path that an application will parse
+
+  std::ofstream parse_file_stream(parse_file_path.c_str());
+
+  if (parse_file_stream.fail())
+  {
+    mooseError("Unable to open file ", parse_file_path);
+  }
+
+  parse_file_stream << this->document_text;
+
+  parse_file_stream.close();
+
+  // copy parent application parameters and modify to set up input check
+
+  InputParameters app_params = moose_app->parameters();
+
+  app_params.set<std::vector<std::string>>("input_file") = {parse_file_path};
+
+  app_params.set<bool>("check_input") = true;
+
```

Comment:
please use empty lines sparingly to separate sections of code

Reply:
Good feedback - I just added a commit addressing this.